crates/nu-command/src/filesystem/glob.rs

```diff
@@ -148,29 +148,54 @@ impl Command for Glob {
             }
         };
 
-        if glob_pattern.item.is_empty() {
-            return Err(ShellError::GenericError {
-                error: "glob pattern must not be empty".into(),
-                msg: "glob pattern is empty".into(),
-                span: Some(glob_pattern.span),
-                help: Some("add characters to the glob pattern".into()),
-                inner: vec![],
-            });
-        }
+        let glob_pattern = match glob_pattern_input {
+            Value::String { val, .. } => {
+                if val.is_empty() {
+                    return Err(ShellError::GenericError {
+                        error: "glob pattern must not be empty".into(),
+                        msg: "glob pattern is empty".into(),
+                        span: Some(glob_span),
+                        help: Some("add characters to the glob pattern".into()),
+                        inner: vec![],
+                    });
+                } else {
+                    val
+                }
+            }
+            Value::Glob { val, .. } => {
+                if val.is_empty() {
+                    return Err(ShellError::GenericError {
+                        error: "glob pattern must not be empty".into(),
+                        msg: "glob pattern is empty".into(),
+                        span: Some(glob_span),
+                        help: Some("add characters to the glob pattern".into()),
+                        inner: vec![],
+                    });
+                } else {
+                    val
+                }
+            }
+            _ => {
+                return Err(ShellError::IncompatibleParametersSingle {
+                    msg: "Incorrect parameter type".to_string(),
+                    span: glob_span,
+                })
```

Comment:
Please no newly invented type errors (pointing exhaustedly at #10698), `IncompatibleParameters` has its proper use when there are mutually exclusive or otherwise incompatible parameters, like extra flags, positionals that make no sense etc.

Reply:
ok, i'll look for a better one. copilot apparently got this one wrong. i need a either this or that error type.

Reply:
Hopefully #13613 helps